tests/bluetooth/controller/common/defaults_cmake.txt

```diff
@@ -41,21 +42,26 @@ FILE(GLOB mock_sources
   ${ZEPHYR_BASE}/tests/bluetooth/controller/mock_ctrl/src/lll.c
   ${ZEPHYR_BASE}/tests/bluetooth/controller/mock_ctrl/src/lll_conn.c
   ${ZEPHYR_BASE}/tests/bluetooth/controller/mock_ctrl/src/ll_assert.c
-  ${ZEPHYR_BASE}/tests/bluetooth/controller/mock_ctrl/src/assert.c
-  ${ZEPHYR_BASE}/tests/bluetooth/controller/mock_ctrl/src/util.c
-  ${ZEPHYR_BASE}/tests/bluetooth/controller/mock_ctrl/src/ticker.c
-  ${ZEPHYR_BASE}/tests/bluetooth/controller/mock_ctrl/src/ull.c
-  ${ZEPHYR_BASE}/tests/bluetooth/controller/mock_ctrl/src/ull_conn_iso.c
-  ${ZEPHYR_BASE}/tests/bluetooth/controller/mock_ctrl/src/ull_peripheral.c
-  ${ZEPHYR_BASE}/tests/bluetooth/controller/mock_ctrl/src/ull_peripheral_iso.c
-  ${ZEPHYR_BASE}/tests/bluetooth/controller/mock_ctrl/src/ull_central.c
-  ${ZEPHYR_BASE}/tests/bluetooth/controller/mock_ctrl/src/ull_scan.c
-  ${ZEPHYR_BASE}/tests/bluetooth/controller/mock_ctrl/src/lll_clock.c
+	${ZEPHYR_BASE}/tests/bluetooth/controller/mock_ctrl/src/util.c
+	${ZEPHYR_BASE}/tests/bluetooth/controller/mock_ctrl/src/ticker.c
+	${ZEPHYR_BASE}/tests/bluetooth/controller/mock_ctrl/src/ull.c
+	${ZEPHYR_BASE}/tests/bluetooth/controller/mock_ctrl/src/ull_conn_iso.c
+	${ZEPHYR_BASE}/tests/bluetooth/controller/mock_ctrl/src/ull_peripheral.c
+	${ZEPHYR_BASE}/tests/bluetooth/controller/mock_ctrl/src/ull_peripheral_iso.c
+	${ZEPHYR_BASE}/tests/bluetooth/controller/mock_ctrl/src/ull_central.c
+	${ZEPHYR_BASE}/tests/bluetooth/controller/mock_ctrl/src/ull_scan.c
+	${ZEPHYR_BASE}/tests/bluetooth/controller/mock_ctrl/src/lll_clock.c
 )
 
 FILE(GLOB common_sources
   ${ZEPHYR_BASE}/tests/bluetooth/controller/common/src/helper_pdu.c
   ${ZEPHYR_BASE}/tests/bluetooth/controller/common/src/helper_util.c
 )
 
+add_definitions(-include kconfig.h)
+if(KCONFIG_OVERRIDE_FILE)
+  add_definitions(-include ${KCONFIG_OVERRIDE_FILE})
+endif()
```

Comment:
we have kconfig support in unittesting so I don't think we should be using this Kconfig override principle.
We should be able to handle this properly in Kconfig files directly, so that unittests are following same pattern as samples.

Please let me know if there are reasons when the common Kconfig approach cannot be used.

Reply:
100% agreed, and the reason for this PR is so that we can get rid of this hack. It is left here because the other unittests rely on this. Once this PR is approved (meaning that we agree on how the framework for the unittests for the controller should be) we can create another PR updating all the other unittests and getting rid of this hack. See #35099